### DISCUSSION TITLE
Brutal interruption of H2 input using VENT

### DISCUSSION THREAD

**Discussion by GuillaumeTHIRIET** [firemodels/fds#13624] [Q&A] [answered]
Dear community member,

I am encountering an issue on a test case I created to test some basic FDS functionalities. 
This is a case involving a VENT flowing H2 gas at rather low velocity into a partially open room with a very simple HVAC system. The FDS code is included below.
[TEST.txt](https://github.com/user-attachments/files/17512244/TEST.txt)

Here a description of my issue,
At the begining of the simulation to 16 seconds, everything seems fine. We can see the H2 flowing in the room and the velocity profile seems correct (at least, nothing weird).

 Then, at the 17 seconds, the flow of H2 seems to be suddenly shut down and the velocity profile brutally change from vertical (buoyancy) to horizontal (no reason ?)

I enclosed some pictures below to show you the difference:
[Pictures.docx](https://github.com/user-attachments/files/17512445/Pictures.docx)

The VENT allowing the H2 to flow in is a 1 cell VENT.

Ps.
I succeeded to solve the problem by increasing the size of the VENT from 1 cell to 4 cells. 
I think that the problem can also be solved by refining the mesh near the VENT but because I am working on a very poor laptop, I did not have tried this solution for now.

**I would like to know if according to your experience, this issue can be due to the **_"1 cell"_** VENT or is this can be due to another element.**

Thank you for your help.

Regards

**Comment by mcgratta** [comment]
I'll check to see if there is an issue with the case. Even if it is crude, you should still achieve basic conservation of mass.

**Reply by GuillaumeTHIRIET** [reply]
Dear Mr. McGrattan,

Thank you evry much for your reply. In any case, I found solutions to bypass this issue and the case I created was mainly to show my new team members what FDS can do in terms of gas dispersion in building (I am trying to introduce FDS in my company).

I am especially looking for a confirmation or a denial of my explaination of this issue (1 cell is not enough for FDS). 

Regards :) (and thanks for your work)

**Reply by mcgratta** [reply, marked as answer]
Sure, a finer grid is always better, but in your case, you should use `MASS_FLUX` on the `SURF` line rather than `MASS_FRACTION` and `VEL`. There is an explanation in the User's Guide. This is particularly important in crude calculations because `MASS_FLUX` will account for the diffusion and convection of the gas. Normally diffusion is small, but on a coarse grid, it is not, and can lead to spurious behavior, like the gas spilling downwards in your case. When you see unphysical effects like this, the first thing to do is consider your grid.

**Reply by GuillaumeTHIRIET** [reply]
Good morning Mr. MacGrattan,

yes, I read about that section in the user's guide (Section 10.1.6 if I am right) but to be honest, I thought that a 20 m.s-1 flow velocity should be enough to counter the harmful influence of diffusion.

I will try another case with this recommendation.

Thank you very much for your help.

Regards

**Reply by GuillaumeTHIRIET** [reply]
Good morning again Mr. MacGrattan,

I confirm that I have far better results using  `MASS_FLUX` instead of `VEL` + `MASS_FRACTION`.

Thank you for that advice

**Comment by mcgratta** [comment]
I added `NO_SLIP=T` to the `SURF` line that defines the hydrogen jet and it prevented the "brutality".  That being said, there is always the possibility of spurious behavior with a coarse mesh.  Also, you can add a line like this
```
&SLCF PBY = 2.0, QUANTITY = 'VOLUME FRACTION', SPEC_ID = 'HYDROGEN', CELL_CENTERED=T /
```
to the input file to see the values of volume fraction that are being computed for each cell. The slices you are using are 8 point averages.
